Source/WebKit/NetworkProcess/cocoa/NetworkTaskCocoa.mm

```diff
@@ -102,6 +103,16 @@ static bool shouldCapCookieExpiryForThirdPartyIPAddress(const WebCore::IPAddress
     return cappedCookies;
 }
 
+#if HAVE(ALLOW_ONLY_PARTITIONED_COOKIES)
+static NSArray<NSHTTPCookie *> *cookiesBySettingPartition(NSArray<NSHTTPCookie *> *cookies, NSString* partition)
+{
+    auto *partitionedCookies = [NSMutableArray arrayWithCapacity:cookies.count];
+    for (NSHTTPCookie *cookie in cookies)
+        [partitionedCookies addObject:WebCore::NetworkStorageSession::setCookiePartition(cookie, partition)];
```

Comment:
It looks like `+cookieWithProperties:` returns a nullable `NSHTTPCookie *`, so we should probably null check this to be safe (instead of crashing).

Reply:
Seems like we need to do a better job of this, in general.